jobs/build-mechanical.Jenkinsfile

```diff
@@ -19,28 +19,30 @@ properties([
 
 node {
     def mechanical_streams = pipeutils.streams_of_type(pipecfg, 'mechanical')
+    def scheduled_streams = pipeutils.scheduled_streams(pipecfg, mechanical_streams)
 
-    change = checkout(
-        [$class: 'GitSCM',
-         userRemoteConfigs: [[url: pipecfg.source_config.url]],
-         branches: pipeutils.streams_as_branches(mechanical_streams)
-        ]
-    )
-
-    if (pipeutils.triggered_by_push()) {
-        stream = pipeutils.stream_from_branch(change.GIT_BRANCH)
-        if (stream in mechanical_streams) {
-            build job: 'build', wait: false, parameters: [
-              string(name: 'STREAM', value: stream)
-            ]
-        }
-    } else {
-        // cron or manual build: build all mechanical streams
-        mechanical_streams.each{
+    if (scheduled_streams) {
```

Comment:
Instead of this, I think we could do something like:

```groovy
def mechanical_streams = ...
def scheduled_streams = ...
if (scheduled_streams) {
    mechanical_streams = scheduled_streams
}

mechanical_streams.each{
    ...
}
```